src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const resObj = {};
+  const arrayWithoutSpaces = sourceString.trim().split(';');
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate original array. In this case, you can chain the `.split(';')` directly with the `.forEach` without storing it in a variable first.